.gitignore

```diff
@@ -1,2 +1,13 @@
-/target
+# Generated by Cargo
+# will have compiled files and executables
+/target/
+
+# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
+# More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
+Cargo.lock
+
+# These are backup files generated by rustfmt
 **/*.rs.bk
+
+# IntelliJ IDEs
+\.idea/
```

Comment:
If IntelliJ isn't inherently tied to the project, I think a local ignore would be cleaner than putting it in the repo's gitignore.

Reply:
Nah, it's pretty easy for people to forget these things from their local gitignore and so the editors for the languages are typically included in the https://github.com/github/gitignore/ -repo's templates